src/common/merge.ts

```diff
@@ -20,12 +25,59 @@ export interface PullRequestDetails {
 const EXPONENTIAL_BACKOFF = 2;
 const MINIMUM_WAIT_TIME = 1000;
 
-const delay = async (duration: number): Promise<void> =>
-  new Promise((resolve: () => void): void => {
+const getIsModified = async (
+  octokit: ReturnType<typeof getOctokit>,
+  query: {
+    pullRequestNumber: number;
+    repositoryName: string;
+    repositoryOwner: string;
+  },
+): Promise<boolean> => {
+  const iterator = makeGraphqlIterator<
```

Comment:
Using an iterator here because the the list should be considered infinite and it's more efficient than a recursive function.